### PR TITLE
Docs: add missing sections, fix formatting/typos

### PR DESCRIPTION
**Pull request by drewrisinger**
### Summary
Tasks: 
* [x] Added missing ``timeline_viewer`` and ``marginal_counts`` sections to the docs
* [x] Formatted docstrings to add types or fix incorrectly-formatted return sections (``disassemble(), marginal_counts(), Parameter, ParameterExpression, QuantumCircuit``)
* [x] Fixed typos in README.md
* [x] Fixed typos in release notes
* [x] **Removed unused ``sphinx-tabs`` plugin.** 
* [ ] Link from ``timeline_drawer()`` release notes to function. This should be working, but isn't
* [x] Maybe include all ``qiskit/visualization/timeline`` documentation?

### Details and comments
Most are straightforward. Tested all changes with a local docs build.

The most questionable change here is probably removing ``sphinx-tabs``, but I couldn't find any references to its features, and docs build cleanly without it. I'm not sure if it's used by downstream Qiskit projects, but it should be moved there if necessary.

### CODE REVIEW

File: qiskit/visualization/__init__.py
Comment (mtreinish) [inline qiskit/visualization/__init__.py:51]: This might be problematic. The issue I faced before when I updated the release notes in #5142 was that `timeline_drawer` as a function doesn't actually exist. The method is `qiskit.visualization.timeline.interface.draw`:

https://qiskit.org/documentation/stubs/qiskit.visualization.timeline.draw.html#qiskit.visualization.timeline.draw

https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/visualization/timeline/interface.py

The issue I faced in that PR was that while `draw` is aliased to `timeline_drawer` on import things don't show up consistently in sphinx. Like when I did this in an attempt locally the link didn't work because it was being translated to `draw` in the build. So I left it alone and decided it was something to fix later. But, lets see what the CI job does here maybe this works and I made some other mistake before (there were a lot of things docs related that I had to fix in that PR).

Comment (mtreinish) [reply qiskit/visualization/__init__.py:51]: Oh, and there is a section for the timeline drawer below we probably should move this too (assuming it works)

File: releasenotes/notes/0.15/add-rep-delay-c97d5aa8fc9696da.yaml
Comment (mtreinish) [inline releasenotes/notes/0.15/add-rep-delay-c97d5aa8fc9696da.yaml:29]: Thanks for updating this here, these reno yaml files are the source of truth for release notes. But for the time being we also have to update these in https://github.com/Qiskit/qiskit/blob/master/docs/release_notes.rst for things to be reflected in the documentation. Hopefully in the near future this duplicate copy won't need to exist, I just need to find time to implement: https://github.com/Qiskit/qiskit/issues/997 Once that's in place just updating the per repo copy is sufficient.

**Review by mtreinish — COMMENTED** [review]
LGTM, thanks for pushing these updates. I'm curious to see if the visualization changes work or not. For the release note changes we'll also have to update things in https://github.com/Qiskit/qiskit/blob/master/docs/release_notes.rst too.

File: qiskit/visualization/__init__.py
Comment (mtreinish) [reply qiskit/visualization/__init__.py:51]: Ok, yeah this works, thanks for doing it. I just downloaded the artifact from the docs ci job and inspected it locally this does the right thing: https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=20562&view=artifacts&type=publishedArtifacts so just move this down to the timeline_drawer section and I think this PR is probably good to merge, unless you had more changes you wanted to make (based on the comments in the PR summary)

**Comment by mtreinish** [comment]
Per the comments in the PR summary I just checked the artifact from the CI build and the timeline_drawer function link for the delay instruction feature release note works as expected. So I think that's good to go. Also I don't think we need to include the entire timeline module in the documentation. I don't think the expectation is to treat anything besides the `draw()` function as a stable api so publishing the docs for them is not something we want to do.

**Comment by mtreinish** [comment]
> The most questionable change here is probably removing sphinx-tabs, but I couldn't find any references to its features, and docs build cleanly without it. I'm not sure if it's used by downstream Qiskit projects, but it should be moved there if necessary.

Yeah this was probably just a copy and paste issue in the configuration from what we used in the metpackage docs. The sphinx tab plugin is used there to build the tabbed interface for the building from source docs: https://qiskit.org/documentation/contributing_to_qiskit.html#installing-terra-from-source it's not used anywhere in terra afaik so it should be safe to remove and simplify things here a bit.

File: releasenotes/notes/0.15/add-rep-delay-c97d5aa8fc9696da.yaml
Comment (drewrisinger) [reply releasenotes/notes/0.15/add-rep-delay-c97d5aa8fc9696da.yaml:29]: Are you asking me to update ``qiskit/qiskit/docs/release_notes.rst``, or will you push that after this?

File: qiskit/visualization/__init__.py
Comment (drewrisinger) [reply qiskit/visualization/__init__.py]: The only thing would be to link the release note from ``qiskit.visualization.timeline_drawer()`` (it currently can't find the function reference) -> ``qiskit.visualization.timeline.draw()``, but I don't know of an easy way to do that via Sphinx. So I'm satisfied that it's in a clearer/more readable state for the purposes of this PR.

File: releasenotes/notes/0.15/add-rep-delay-c97d5aa8fc9696da.yaml
Comment (mtreinish) [reply releasenotes/notes/0.15/add-rep-delay-c97d5aa8fc9696da.yaml:29]: I was asking if you would push a follow up PR to make the changes there in `qiskit/qiskit`. But if you'd prefer I can do it too.

Comment (drewrisinger) [reply releasenotes/notes/0.15/add-rep-delay-c97d5aa8fc9696da.yaml:29]: I can do that once this is merged 👍

**Review by mtreinish — APPROVED** [review]

**Pull request merged**
